webui.py

```diff
@@ -900,22 +916,55 @@ def process_images(
     return output_images, seed, info, stats
 
 
+def txt2img_range_based_valid(steps_start, steps_end, cfgs_start, cfgs_end, denoise_start, denoise_end):
+    if steps_start <= steps_end:
+        if cfgs_start <= cfgs_end:
+            if denoise_start <= denoise_end:
+                return True
+    print(f"Range based gen paramterers invalid:\n\tsteps {steps_start}-{steps_end}\n\tcfgs {cfgs_start}-{cfgs_end}\n\tdenoise {denoise_start}-{denoise_end}")
+    return False
+
+def str_to_int_or_val(s, dval):
+    if isinstance(s, str):
+        try:
+            return int(s)
+        except:
+            return dval
+    return dval
+
+def str_to_float_or_val(s, dval):
+    if isinstance(s, str):
+        try:
+            return float(s)
+        except:
+            return dval
+    return dval
+    
 def txt2img(prompt: str, ddim_steps: int, sampler_name: str, toggles: List[int], realesrgan_model_name: str,
             ddim_eta: float, n_iter: int, batch_size: int, cfg_scale: float, seed: Union[int, str, None],
-            height: int, width: int, fp):
+            height: int, width: int, fp,
+            rb_steps_start: Union[int, str, None], rb_steps_end:Union[int, str, None], rb_cfgs_start:Union[float, str, None], rb_cfgs_end:Union[float, str, None]):
+    
     outpath = opt.outdir_txt2img or opt.outdir or "outputs/txt2img-samples"
     err = False
     seed = seed_to_int(seed)
 
+
+    rb_steps_start = str_to_int_or_val(rb_steps_start, txt2img_defaults["rb_steps_start"])
+    rb_steps_end = str_to_int_or_val(rb_steps_end, txt2img_defaults["rb_steps_end"])
+    rb_cfgs_start = str_to_float_or_val(rb_cfgs_start, txt2img_defaults["rb_cfgs_start"])
+    rb_cfgs_end = str_to_float_or_val(rb_cfgs_end, txt2img_defaults["rb_cfgs_end"])
+
     prompt_matrix = 0 in toggles
     normalize_prompt_weights = 1 in toggles
     skip_save = 2 not in toggles
     skip_grid = 3 not in toggles
     sort_samples = 4 in toggles
     write_info_files = 5 in toggles
     jpg_sample = 6 in toggles
-    use_GFPGAN = 7 in toggles
-    use_RealESRGAN = 8 in toggles
+    range_based_gen = 7 in toggles
+    use_GFPGAN = 8 in toggles
+    use_RealESRGAN = 8 in toggles if GFPGAN is None else 9 in toggles # possible index shift

```

Comment:
Just change the numbers, there's no need to have if else, I keep removing it and you (or someone else without me noticing on merge) keeps adding it back in